### DISCUSSION TITLE
What does the mix node do in the default MOL_atomic_material?

### DISCUSSION THREAD

**Discussion by PlethoraChutney** [BradyAJohnston/MolecularNodes#117] [Q&A] [answered]
Hello! I'm wondering what the purpose of this cluster of nodes is in the default MOL_atomic_material:

<img width="270" alt="image" src="https://user-images.githubusercontent.com/27385979/208515283-f48abce6-fd98-4783-8a1c-d01a67329ebe.png">

Thank you!

**Comment by BradyAJohnston** [comment, marked as answer]
For mesh / points (Surface style, Atoms Cycles & ribbon) the colour information is taken from the Geometry (top 'Attribute' node). For the styles that involve instancing of geometry like Ball and Stick & Atoms EEVEE, the instance needs to get its colour information from the Instanced instead of the geometry. 

This is my 'elegant' (not at all) solution to mix between the two based on if the it is an instance or not. Is the instances absolute position is > 0, then it gets the colour from the Instancer otherwise it gets it from the geometry. 

I certainly don't like this setup as is, but it's the best that I could come up with at the time to solve that little problem. Thinking about it, it's potentially best to have a separate material for EEVEE atoms etc than for cycles atoms. I've certainly run into the problem of wanting different materials myself

**Comment by PlethoraChutney** [comment]
Ahh, I see! That explains why deleting one seems to not change the color --- it's only ever using one at a time. Thanks!
